solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/XMLMessageMapper.java

```diff
@@ -225,7 +228,16 @@ public XMLMessage map(Message<?> message, Collection<String> excludedHeaders, bo
 			if (!header.getValue().isReadable()) {
 				continue;
 			}
-			builder.setHeaderIfAbsent(header.getKey(), header.getValue().getReadAction().apply(xmlMessage));
+			if (header.getKey().equals(SolaceHeaders.DELIVERY_COUNT) && !readDeliveryCount.get()) {
+				continue;
+			}
+			try {
+				builder.setHeaderIfAbsent(header.getKey(), header.getValue().getReadAction().apply(xmlMessage));
+			} catch (SolaceDeliveryCountException e) {
+				logger.info(String.format("Failed to retrieve delivery count. Will not re-attempt until flow rebinds or reconnects. Error: %s", e.getMessage()));
+				readDeliveryCount.set(false);
+				continue;
+			}
```

Comment:
As per previous comments, this should be handled generically, so something like this should be enough:
```java
try {
builder.setHeaderIfAbsent(header.getKey(), header.getValue().getReadAction().apply(xmlMessage));
} catch {UnsupportedOperationException e} {
// ...
}
```